lib/rack/lint.rb

```diff
@@ -318,8 +332,10 @@ def check_environment(env)
         end
 
         ## The <tt>HTTP_HOST</tt> must be a valid authority as defined by RFC7540.
-        unless (URI.parse("http://#{env[HTTP_HOST]}/") rescue false)
-          raise LintError, "#{env[HTTP_HOST]} must be a valid authority"
+        if http_host = env[HTTP_HOST]
```

Comment:
Not sure if this is used elsewhere.  Does this result in any verbose mode unused variable warnings?

Reply:
I checked it does not appear to emit any warning.

Reply:
That's because you changed the code to use the variable in the error message (but not in the URI parse string for some reason).

When I reviewed this, the variable wasn't used: https://github.com/rack/rack/commit/9351e88bf4f55772309e391e336ad1f654df7c52#diff-2db485952a203ded9d5ae1301e35d82601b1c2410f914e767b02c4933b7e8a57R337

Reply:
Thanks, fixed.